linq4j/src/main/java/org/apache/calcite/linq4j/tree/Expressions.java

```diff
@@ -2829,6 +2869,18 @@ public static ThrowStatement throw_(Expression expression) {
     return new ThrowStatement(expression);
   }
 
+  /**
+   * Creates an expression that represents the throwing of an exception.
+   */
+  public static Expression throwExpr(Expression expression) {
```

Comment:
There's a ThrowStatement.
Maybe having an expression is useful, since it can be used in an expression context - you don't need to break the current expression to create a new statement.
I would document this feature.